modules/persistence/src/services/metadata/associated_products/index.ts

```diff
@@ -74,10 +82,25 @@ export const hasAssociations = (metadata) => !!metadata.associated_products?.len
 const umbrellaMetadataEntry = async (project: string): Promise<Metadata> => {
   try {
     const snooty = await db();
+
+    // first find any umbrella
+    const umbrella = await snooty.collection('metadata').findOne({
+      'associated_products.name': project,
+      is_merged_toc: false,
+    });
+
+    if (!umbrella) {
+      return null as unknown as Metadata;
+    }
+
+    const repoDoc = await getRepoBranchesEntry(project);
```

Comment:
Did we want to use the `branch` from umbrella as an argument here?

Reply:
i think this would re-introduce the error here. i purposely left out the branch here

ie. `writers staging branch (which includes associated products) is persisted the latest` ->  no repo_branches entry for this branch